src/dataflow/src/source/postgres.rs

```diff
@@ -353,9 +353,16 @@ impl SimpleSource for PostgresSourceReader {
 
         loop {
             match self.produce_replication(timestamper).await {
-                Ok(_) => log::info!("replication interrupted with no error"),
+                Ok(_) => log::info!(
+                    "replication for source {} interrupted with no error",
```

Comment:
What would "interrupted with no error mean"?

Reply:
In theory, it means that the replication process in `produce_replication` was interrupted somehow, but that there wasn't a `Recoverable` or `Fatal` error. This doesn't seem to ever happen in practice (nothing in `produce_replication` returns `Ok(...)`), but maybe there's a future where it could? Alternatively, we could make it `unreachable`? @petrosagg

Reply:
I think you're right, we should mark it as `unreachable!()`